### DISCUSSION TITLE
HTTP redirect with an app/button in UI

### DISCUSSION THREAD

**Discussion by rei-ber** [datalust/seq-tickets#1778] [Help] [answered]
Hey people,

if I found an interesting event in seq, I want to click on a button to redirect to another endpoint with some event attributes in query parameter. Assume I have the following event:

```json
{
"id": 1,
"loglevel": "error"
}
```

With the app `Seq.App.HttpRequest` I can send a request to a http server successfully. I configured the url in the app settings as `http://localhost:8000/post?id={id}&loglevel={loglevel}` which ends in `http://localhost:8000/post?id=1&loglevel=error`. That's a helpful app. 

But the idea is, to click on a link in seq UI and get redirected to the url `http://localhost:8000/post?id=1&loglevel=error` instead of posting a request to it. The idea is, that a new browser tab will open with this url.

Is there any possibility to do that? Sure, I can write my own app for this, but I am not familiar with c#. Is there a chance to write an app in python as well?

Kind regards
Fabian

**Comment by liammclennan** [comment, marked as answer]
Hi @rei-ber,

We have refrained from allowing the injection of links into the Seq UI because of the security implications. 

It _might_ be possible to do this using CSS Style overrides (Settings > Instance), although I don't think so, because CSS can only read attributes, not text content, and of course all of the security concerns still apply.

**Reply by rei-ber** [reply]
Hi @liammclennan 

thanks for your response! Too bad, that it is not possible. And CSS is not an option, you are right. But I understand the reason to not provide this option.

Is it an idea to have this CSS override feature for javascript to add more individual functionality to Seq? I mean, to have the same kind of CSS box under Settings -> Instance, but for javascript code to add?

**Reply by liammclennan** [reply]
Hi @rei-ber,

You can do this now with greasemonkey or similar browser extensions. Note that this would make supporting the product very difficult.

**Reply by rei-ber** [reply]
Yes that's true; we already discussed this option.

Thanks for your help!
